app/models/report.rb

```diff
@@ -46,4 +46,11 @@ def end_date
       start_date + duration.days
     end
   end
+
+  private
+
+  def active_components(opts={})
+    result = text_components.includes(:channels)
+    text_components.select {|c| c.active?(opts) }
+  end
```

Comment:
I think generally we will only need active components. However, it’s definitely  a good idea to have an option to include unpublished components as well, in case we need to preview them before publishing in the future. Should default to published text components only, though.

In my opinion, the benefits of a more generally applicable method justify the work that has to be done in order to accomplish this, even if that feature isn’t currently on our list.